wgpu-types/src/lib.rs

```diff
@@ -6520,6 +6540,7 @@ impl Default for InstanceDescriptor {
     fn default() -> Self {
         Self {
             backends: Backends::all(),
+            flags: InstanceFlags::default(),
```

Comment:
I feel like we should use `from_build_type` here to keep the current behavior.

Reply:
Having default return a different thing depending on the build configuration sort of rubs me the wrong way but I don't feel very strongly about it so I changed `InstanceFlag`'s default impl to do `from_build_config` instead.

Reply:
> Having default return a different thing depending on the build configuration sort of rubs me the wrong way but I don't feel very strongly about it

That's a good point. I was mainly worried that we won't get debug info and validation errors by default in debug builds (which was the default behavior before).

Not sure what's best here since now wgpu users will have to pass the flags.

Reply:
DIscussed in call - keeping behavior the same.